enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/PipelineInformation.scala

```diff
@@ -181,8 +181,8 @@ class PipelineInformation(private var slots: Map[String, Slot],
     case _ => throw new InternalException("Uh oh... There was no slot for `$name`")
   }
 
-  def foreachSlot[U](f: ((String,Slot)) => U): Unit =
-    slots.foreach(f)
+  def foreachSlotOrdered[U](f: ((String, Slot)) => U): Unit =
+    slots.toSeq.sortBy(_._2)(SlotOrdering).foreach(f)
```

Comment:
I think you should use `toIndexedSeq` or `toVector` here, sorting a `scala.List` is really slow.